Main branch fails to build

A build today fails with " [ERROR] TestEngine with ID 'junit-jupiter' failed to discover tests" as can be seen at https://ci.eclipse.org/tm4e/job/TM4E/job/main/349/console .
Latest merged PR is https://github.com/eclipse-tm4e/tm4e/pull/863 but even after reverting it locally build fails in the same way. 
Using -Dtm4e.target-platform=latest makes the build work and changing https://github.com/eclipse-tm4e/tm4e/blob/1d2220d603ff17e42836dab3e59368c204e786b6/target-platforms/oldest.target#L13 to 4.30 also fixes it. 

IMO 4.29 is too old to handle some content in 2025-03 Orbit ( https://github.com/eclipse-tm4e/tm4e/blob/1d2220d603ff17e42836dab3e59368c204e786b6/target-platforms/oldest.target#L24 ) and this breaks JUnit .
I would propose using newer platform version as once this kind of issue pops up it only becomes worse over time.

--- a/target-platforms/latest.target
+++ b/target-platforms/latest.target
@@ -16,8 +16,6 @@
       <unit id="org.eclipse.ui.tests.harness" version="0.0.0"/>
       <unit id="org.eclipse.jdt.annotation" version="0.0.0"/>
       <unit id="org.eclipse.jdt.junit5.runtime" version="0.0.0"/>
-      <unit id="org.junit" version="0.0.0"/>
-      <unit id="junit-jupiter-api" version="0.0.0"/>
     </location>
 
     <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
@@ -34,6 +32,8 @@
       <unit id="org.joni.source" version="0.0.0"/>
       <unit id="org.snakeyaml.engine" version="0.0.0"/>
       <unit id="org.snakeyaml.engine.source" version="0.0.0"/>
+      <unit id="org.junit" version="0.0.0"/>
+      <unit id="junit-jupiter-api" version="0.0.0"/>
     </location>
   </locations>
 

--- a/target-platforms/oldest.target
+++ b/target-platforms/oldest.target
@@ -16,8 +16,6 @@
       <unit id="org.eclipse.ui.tests.harness" version="0.0.0"/>
       <unit id="org.eclipse.jdt.annotation" version="0.0.0"/>
       <unit id="org.eclipse.jdt.junit5.runtime" version="0.0.0"/>
-      <unit id="org.junit" version="0.0.0"/>
-      <unit id="junit-jupiter-api" version="0.0.0"/>
     </location>
 
     <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
@@ -34,6 +32,8 @@
       <unit id="org.joni.source" version="0.0.0"/>
       <unit id="org.snakeyaml.engine" version="0.0.0"/>
       <unit id="org.snakeyaml.engine.source" version="0.0.0"/>
+      <unit id="org.junit" version="0.0.0"/>
+      <unit id="junit-jupiter-api" version="0.0.0"/>
     </location>
   </locations>
 

--- a/target-platforms/unstable.target
+++ b/target-platforms/unstable.target
@@ -16,8 +16,6 @@
       <unit id="org.eclipse.ui.tests.harness" version="0.0.0"/>
       <unit id="org.eclipse.jdt.annotation" version="0.0.0"/>
       <unit id="org.eclipse.jdt.junit5.runtime" version="0.0.0"/>
-      <unit id="org.junit" version="0.0.0"/>
-      <unit id="junit-jupiter-api" version="0.0.0"/>
     </location>
 
     <location includeAllPlatforms="false" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
@@ -34,6 +32,8 @@
       <unit id="org.joni.source" version="0.0.0"/>
       <unit id="org.snakeyaml.engine" version="0.0.0"/>
       <unit id="org.snakeyaml.engine.source" version="0.0.0"/>
+      <unit id="org.junit" version="0.0.0"/>
+      <unit id="junit-jupiter-api" version="0.0.0"/>
     </location>
   </locations>
 

